assets/training/distillation/components/data_generation/src/generate_data.py

```diff
@@ -353,7 +305,13 @@ def process_conversational_request(idx: str, data: dict, url: str, endpoint_key:
             synthetic_responses = []
             for message in messages:
                 role = message['role']
-                if role in ('system', 'user'):
+                if role == 'system':
```

Comment:
Push the conditional implementation to a function. This way, any changes even on a later date will be limited to the function.

Reply:
Done